lib/src/sv_openssl.c

```diff
@@ -147,6 +147,51 @@ openssl_private_key_malloc(sign_or_verify_data_t *sign_data,
   return status;
 }
 
+/**
+ * Extracts the private key from the signing data structure.
+ * This function retrieves the private key stored within the `sign_data` structure
+ * and returns it as a dynamically allocated null-terminated string in PEM format.
+ */
+char *
+get_private_key_from_sign_data(sign_or_verify_data_t *sign_data)
```

Comment:
The only thing this function uses from `sign_data` is the member `key`. May I suggest that this function takes that key as input instead.

Then rename the function accordingly.

And the function name should begin with `openssl_`

And also, the declaration should be in sv_openssl_internal.h and not in sv_internal.h